runtime/context.go

```diff
@@ -119,9 +119,10 @@ func WithEndpointRequestHeadersField(ctx context.Context, requestHeaders map[str
 
 // GetEndpointRequestHeadersFromCtx returns the endpoint request headers, if it exists on context
 func GetEndpointRequestHeadersFromCtx(ctx context.Context) map[string]string {
-	requestHeaders := make(map[string]string)
+	requestHeaders := make(map[string]string, 0)
```

Comment:
nit: 0 is default length hence not required to specify

Reply:
See https://pkg.go.dev/builtin#make . Go allocates a small size, if it is not specified.